kotlin-native/runtime/src/launcher/cpp/launcher.cpp

```diff
@@ -30,8 +31,14 @@ using namespace kotlin;
 //--- Setup args --------------------------------------------------------------//
 
 OBJ_GETTER(setupArgs, int argc, const char** argv) {
+  if (argc > 0 && argv[0][0] != '\0') {
+    // Don't set the programName to an empty string (by checking argv[0][0] != '\0') to make all platforms behave the same:
+    // Linux would set argv[0] to "" in case no programName is passed, whereas Windows & macOS would set argc to 0.
+    kotlin::programName = strndup(argv[0], 4096);
```

Comment:
Funny thing this code doesn't compile on mingw_x64. You can observe this by running
```
./gradlew clangFrontendLauncherMainMingw_x64
```

Apparently `strndup` is not available in (our version of) MinGW.

Reply:
Thats really interesting. I therefore reverted to the old `kotlin::programName = argv[0];` implementation. Those tasks compile then successfully: `./gradlew clangFrontendLauncherMainMingw_x64 clangFrontendLauncherMainAndroid_arm64 clangFrontendLauncherMainMacos_x64 clangFrontendLauncherMainLinux_x64`.